server/provider/requests.go

```diff
@@ -0,0 +1,29 @@
+package provider
+
+import "errors"
+
+func (provider *networkProvider) getResource(url string, response resourceApiResponse) error {
```

Comment:
I've separated the functionality this way (different files), so that it's easier to reason about it as a "future-to-be-separate-component". At the moment, it still belongs to `networkProvider`, but added a (small) drop of cohesion.

Reply:
yes. Maybe a refactor work in the future, as you stated, to separate in smaller components